src/posts/abstracting-the-frontend/index.md

```diff
@@ -0,0 +1,124 @@
+---
+path: '/blog/abstracting-the-front-end'
+title: 'Abstracting The Front End'
+date: '2020-03-15'
+featuredImage: './cover.jpg'
+---
+
+We so rarely talk about the front end in abstract terms. It's all _"how to x in React"_ or _"Scaling apps in my favourite framework"._
+
+Let's take a moment and talk about how we could split up our front-end concerns on a higher level,
+agnostic of any framework or technology, so that we can avoid distractions when dealing with the aforementioned challenges.
+
+### The Front End
+
+What is the front end?
+
+It is my opinion that today's front-end developer should harness the skill to provide a user with a visual interface
+to interact with the product one is working on. Be it a web application, a mobile app, a car's dashboard or a kitchen fridge.
+This is inherently the job of the front-end developer.
+
+We can safely assume that the front end can appear in many forms, sizes and places, and all these require their own unique
+behaviour in order to fulfill the needs of our users.
+
+As an organisation grows, it is almost certain that the product it provides will need to be presented in more than one of these
+types of platforms. For example, a media streaming service will effectively want to be present in all of them,
+perhaps even the fridge!
+
+### Separation of concerns
+
+Software architects often like to use layers to sort the different responsibilities of a program into distinct,
+easy-to-reason-about chunks that can be stacked to form a hierarchy. I will attempt to do the same thing with what I believe
+to be the chunks that a front end should consist of, so that we can maintain the same level of flexibility and productivity
+across all the types of gadgets and gizmos that exist out there.
+
+_**Quick disclaimer:** Perhaps these concepts can be sorted into something more pragmatic than layers,
```

Comment:
Tone down your defensiveness 15% - you should have established why you think I should listen to you, now say what you have to say as clearly as you can. That there might be better ways, especially for me, goes without saying.

Reply:
I agree that this is too defensive. I think I will look over this section and try to make fewer statements about the work of software architects. I also think there is a better way of modelling this in a way that does not require excuses.